src/fsharp/CompileOps.fs

```diff
@@ -1835,6 +1835,67 @@ let ComputeMakePathAbsolute implicitIncludeDir (path: string) =
     with 
         :? System.ArgumentException -> path  
 
+//
+// Add your features to this List - in code use languageVersion.SupportsFeature(LanguageFeatures.yourFeature) 
+// a return value of false means your feature is not supported by the users language selection
+// All new language features added from now on must be protected by this.
+// Note:
+//   *  The fslang design process will require a decision about feature name and whether it is required.
+//   *  When a feature is assigned a release language, we will scrub the code of feature references and apply
```

Comment:
I don't understand this. What if a user explicitly sets LangVersion to a lower one that doesn't support the feature? Will the code fail to compile?

Also I feel like we'll fail immediately at doing this:

> we will scrub the code of feature references

Nothing against us personally, just feels like the nature of needing to go back and clean up after something is implemented means we'll miss stuff.

Reply:
Furthermore, wouldn't we just stick everything behind `preview` unless we've explicitly merged in a feature that we have no intention of shipping in the next language version?